### PR TITLE
Connect to API

### PR DESCRIPTION
**Pull request by ankitt26**
In this branch, I did-
- use of the [Bookstore API](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739) as your backend.
- Use [axios](https://axios-http.com/docs/intro) with [createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) to make your HTTP requests to the backend.
- Fetch your books from the Bookstore API after the component has rendered
- Refactor your add book and remove book features to persist your changes in the server

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -1,41 +1,65 @@
...
+    const apiResult = response.data.result;
+    return { apiResult, book };
+  } catch (error) {
+    if (axios.isAxiosError(error)) {
+      throw new Error(error.response.data.message);
+    } else {
+      throw error;
+    }
+  }
+});
```
Comment (nahnahdev) [inline src/redux/books/booksSlice.js]: - [x] Great job so far @ankitt26 There are just a few changes you need to make and you'll be on your way to the next Milestone :+1: 
- [ ] I noticed when I was testing your app that the add, and remove functionalities do work as expected. However, you have to refresh the page each time to see if a book was added or removed :100: 
- [ ] This behavior is not ideal for a user-friendly experience. I suggest you ensure that the changes are reflected immediately without requiring a page refresh.  :100: 

[Screencast from 06-16-2023 05:27:11 PM.webm](https://github.com/ankitt26/Bookstore/assets/102699958/2f9a3928-4f69-4630-bea9-f36e352830be)

**Review by nahnahdev — CHANGES_REQUESTED** [review]
## Project Status: CHANGES REQUIRED :recycle: 
Hi @ankitt26 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights :fire: 
- The `addBook` and `removeBook` features work the same as before :heavy_check_mark: 
- All HTTP requests are made using Axios :heavy_check_mark: 
- `fetchBooks` function updates the state of booksSlice :heavy_check_mark: 
- `UseEffect` is used when dispatching the fetchBooks function in `<BookList>` :heavy_check_mark: 
- `CreateAsyncThunk` is imported into the project :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by abdulhamiid — APPROVED** [review]
# :green_circle:  Approved! :partying_face:
Hi @ankitt26 👋, 
image
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 🌟

- ✔️Understand the concept of state management.
- ✔️Use store, actions, and reducers in React.
- ✔️Connect React and Redux.
- ✔️Connect an existing API via Redux and use the data to display something in a UI.

## Optional suggestions 🤷‍♂️
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ankitt26** [comment]
Thanks for reviewing 😊.

**Pull request merged**